proteus/NonlinearSolvers.py

```diff
@@ -755,6 +755,8 @@ def solve(self,u,r=None,b=None,par_u=None,par_r=None):
         self.F.secondCallCalculateResidual = 0
         self.computeResidual(u,r,b)
         u[:] = r
+        if par_u is not None:
```

Comment:
@ejtovar this is really the only line I have a question about. This is the special SW nonlinear solver, but I don't know why these lines would be in there if you didn't put them in a commit at some point.